scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1061,13 +1086,15 @@ def update_cache( self, run_state ):
 
             try:
                 current_time = time.time()
-                if local_state.batch_pod_updates:
-                    self._pods.update( local_state.k8s, local_state.node_filter, 'Pod' )
+                if not self._state.cache_config.use_controlled_warmer and local_state.batch_pod_updates:
+                    self._pods_cache.update(local_state.k8s, local_state.node_filter, 'Pod')
                 else:
                     global_log.log( scalyr_logging.DEBUG_LEVEL_1, "Purging unused pods" )
-                    self._pods.purge_expired( access_time=current_time )
+                    self._pods_cache.purge_expired(access_time=current_time)
```

Comment:
controlled warmer code path should also use the same non-batched expiration logic.